tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,267 @@
 COPR_URL = 'https://copr.fedorainfracloud.org/coprs'
 
 
+class InstallBase(tmt.utils.Common):
+    """ Base class for installation implementations """
+
+    def __init__(
+            self,
+            parent,
+            guest,
+            package,
+            pkg_data,
+            pkg_mgr="",
+            plugin='dnf-plugins-core'):
+        """ Initialize installation data """
+        super().__init__(parent=parent)
+        self.local_packages = []
+        self.debuginfo_packages = []
+        self.repo_packages = []
+        self.pkg_data = pkg_data
+        self.exclude = self.pkg_data.get('exclude', [])
+        self.guest = guest
+        self.package = package
+        self.pkg_mgr = pkg_mgr
+        self.plugin = plugin
+        self.check_sanity()
+        self.process_packages()
+        self.prepare_sudo()
+
+    def check_sanity(self):
+        """ Check parameters, bail out if nothing to do """
+        # Check rpm packages in local directories
+        self.packages = self.pkg_data.get('package', [])
+        self.directories = self.pkg_data.get('directory', [])
+        if not len(self.packages) > 0 and not len(self.directories) > 0:
```

Comment:
`if not self.packages and not self.directories:` would be simpler and do the same job.